### PR TITLE
EFRS-778-logged-user-info-add

### PR DESCRIPTION
**Pull request by omorgunova-pl**

### CODE REVIEW

**Review by isobolev — COMMENTED** [review]
please refactor to container and presentationals components

File: ui/src/app/features/tool-bar/tool-bar.component.html [hunk 1]
```diff
@@ -19,18 +19,21 @@
...
+        <button mat-icon-button [matMenuTriggerFor]="menu">
+          <img [src]="userAvatarInfo$ | async" />
+        </button>
+      </span>
+    </div>
     <mat-menu #menu="matMenu">
-<!--      <button mat-menu-item>-->
-<!--        <span>Account</span>-->
-<!--      </button>-->
+      <!--      <button mat-menu-item>-->
```
Comment (isobolev) [inline ui/src/app/features/tool-bar/tool-bar.component.html:31]: please remove commented code

File: ui/src/app/features/tool-bar/tool-bar.component.ts [hunk 2]
```diff
@@ -43,6 +45,14 @@ export class ToolBarComponent implements OnInit {
       this.isAuthenticated = state.isAuthenticated;
       this.user = state.user;
     });
+    this.getUserInfo();
+  }
+
+  getUserInfo() {
+    this.userInfoService.get().subscribe((data: any) => {
+      this.userInfo = (data.firstName + ' ' + data.lastName);
+      console.log(this.userInfo);
```
Comment (isobolev) [inline ui/src/app/features/tool-bar/tool-bar.component.ts:54]: no console.log

File: ui/src/app/features/tool-bar/tool-bar.component.ts [hunk 3]
```diff
@@ -43,6 +45,14 @@ export class ToolBarComponent implements OnInit {
       this.isAuthenticated = state.isAuthenticated;
       this.user = state.user;
     });
+    this.getUserInfo();
+  }
+
+  getUserInfo() {
+    this.userInfoService.get().subscribe((data: any) => {
```
Comment (isobolev) [inline ui/src/app/features/tool-bar/tool-bar.component.ts:52]: memory leak, please refactor to container and presentational component

File: ui/src/app/features/tool-bar/tool-bar.component.ts [hunk 4]
```diff
@@ -43,6 +45,14 @@ export class ToolBarComponent implements OnInit {
       this.isAuthenticated = state.isAuthenticated;
```
Comment (isobolev) [inline ui/src/app/features/tool-bar/tool-bar.component.ts:45]: please refactor to get this.isAuthenticated and this.user from selectors

**Comment by omorgunova-pl** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by omorgunova-pl** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by omorgunova-pl** [comment]
recheckcla

**Comment by andreigaevsky** [comment]
recheckcla

**Comment by andreigaevsky** [comment]
recheckcla

File: ui/src/app/features/tool-bar/tool-bar.component.ts [hunk 5]
```diff
@@ -14,39 +14,24 @@
...
-  user = null;
+  @Input() userAvatarInfo: string;
+  @Input() userName: string;
+  @Output() logout = new EventEmitter();
 
-  constructor( private store: Store<AppState>) {
-    this.getState$ = this.store.select(selectAuthState);
-    this.userAvatarInfo$ = this.store.select(selectUserAvatar);
-  }
+  constructor() {}
```
Comment (isobolev) [inline ui/src/app/features/tool-bar/tool-bar.component.ts:30]: please remove empty constructor

File: ui/src/app/store/index.ts [hunk 6]
```diff
@@ -20,6 +20,7 @@ import {RouterStateUrl} from './router/reducer';
 import {UserInfoReducer, UserInfoState} from './userInfo/reducers';
 
 export interface AppState {
+  isAuthenticated: any;
```
Comment (isobolev) [inline ui/src/app/store/index.ts:23]: not used property

File: ui/src/app/store/userInfo/selectors.ts [hunk 7]
```diff
@@ -34,3 +34,13 @@ export const selectUserAvatar = createSelector(
   selectUserInfoState,
   (userInfo) => userInfo.avatar || 'assets/img/avatar.png'
 );
+
+export const selectUserName = createSelector(
+  selectUserInfoState,
+  (userInfo) => userInfo.firstName + ' ' + userInfo.lastName
+);
+
+export const isAuthenticatedState = createSelector(
```
Comment (isobolev) [inline ui/src/app/store/userInfo/selectors.ts:43]: not used selector

**Review by isobolev — COMMENTED** [review]
nice work, looks much better, just minor comments left

File: ui/src/app/features/tool-bar/tool-bar.component.ts [hunk 8]
```diff
@@ -14,39 +14,24 @@
...
 
-  ngOnInit() {
-    this.getState$.subscribe((state) => {
-      this.isAuthenticated = state.isAuthenticated;
-      this.user = state.user;
-    });
-  }
+  ngOnInit() {}
```
Comment (isobolev) [inline ui/src/app/features/tool-bar/tool-bar.component.ts:32]: please remove empty ngOnInit as well

**Comment by omorgunova-pl** [comment]
recheckcla

**Review by isobolev — APPROVED** [review]

**Pull request merged**